cirq/google/calibration/phased_fsim.py

```diff
@@ -130,3 +140,217 @@ def override_by(self, other: 'PhasedFSimCharacterization') -> 'PhasedFSimCharact
+        """Magic method for the JSON serialization protocol."""
+        result_dict = super()._json_dict_()
+        result_dict['cirq_type'] = 'FloquetPhasedFSimCalibrationResult'
+        result_dict['options'] = self.options
+        return result_dict
+
+
+@dataclasses.dataclass(frozen=True)
+class FloquetPhasedFSimCalibrationRequest(PhasedFSimCalibrationRequest):
+    """PhasedFSim characterization request specific to Floquet calibration.
+
+    Attributes:
+        options: Floquet-specific characterization options.
+    """
+
+    options: FloquetPhasedFSimCalibrationOptions
+
+    def to_calibration_layer(self) -> CalibrationLayer:
+        circuit = Circuit([self.gate.on(*pair) for pair in self.pairs])
+        return CalibrationLayer(
+            calibration_type=_FLOQUET_PHASED_FSIM_HANDLER_NAME,
+            program=circuit,
+            args={
+                'est_theta': self.options.characterize_theta,
+                'est_zeta': self.options.characterize_zeta,
+                'est_chi': self.options.characterize_chi,
+                'est_gamma': self.options.characterize_gamma,
+                'est_phi': self.options.characterize_phi,
+                'readout_corrections': True,
+            },
+        )
+
+    # TODO: Handle unsuccessful calibrations and throw appropriate exceptions.
+    def parse_result(self, result: CalibrationResult) -> PhasedFSimCalibrationResult:
+        decoded: Dict[int, Dict[str, Any]] = collections.defaultdict(lambda: {})
+        for keys, values in result.metrics['angles'].items():
+            for key, value in zip(keys, values):
+                match = re.match(r'(\d+)_(.+)', str(key))
+                if not match:
+                    raise ValueError(f'Unknown metric name {key}')
+                index = int(match[1])
+                name = match[2]
+                decoded[index][name] = value
+
+        parsed = {}
+        for data in decoded.values():
+            a = v2.qubit_from_proto_id(data['qubit_a'])
+            b = v2.qubit_from_proto_id(data['qubit_b'])
+            parsed[(a, b)] = PhasedFSimCharacterization(
+                theta=data.get('theta_est', None),
+                zeta=data.get('zeta_est', None),
+                chi=data.get('chi_est', None),
+                gamma=data.get('gamma_est', None),
+                phi=data.get('phi_est', None),
+            )
+
+        return FloquetPhasedFSimCalibrationResult(
+            parameters=parsed, gate=self.gate, options=self.options
+        )
+
+    @classmethod
+    def _from_json_dict_(
+        cls,
+        gate: Gate,
+        pairs: List[Tuple[Qid, Qid]],
+        options: FloquetPhasedFSimCalibrationOptions,
+        **kwargs,
+    ) -> 'PhasedFSimCalibrationRequest':
+        """Magic method for the JSON serialization protocol.
+
+        Converts serialized dictionary into a dict suitable for
+        class instantiation."""
+        instantiation_pairs = tuple((entry[0], entry[1]) for entry in pairs)
```

Comment:
why not `tuple(pairs)`? is `entry` larger than two entries? maybe tuple unpacking could make this explicit (if it's true)

```python
tuple((a, b) for a, b, _ in pairs)
```

Reply:
Done.